addons/mail/static/src/model/model_field.js

```diff
@@ -285,9 +320,14 @@ class ModelField {
      *
      * @param {mail.model} record
      * @param {any} newVal
+     * @param {Object} [param2={}]
+     * @param {boolean} [param2.hasToUpdateInverse=true] whether updating the
+     *  current field should also update its inverse field. Only applies to
+     *  relational fields. Typically set to false only during the process of
+     *  updating the inverse field itself, to avoid unnecessary recursion.
      * @returns {boolean} whether the value changed for the current field
      */
-    set(record, newVal) {
+    set(record, newVal, { hasToUpdateInverse = true } = {}) {
```

Comment:
It's a bit disturbing to see explicit `hasToUpdateInverse` a bit everywhere.

I think if we use a named positional argument `options` and pass `options`, everything becomes much more readable.

_(extra note: of course, default options can't just use ES6 default params, and also prefer not modifying parameter `options` in place, because it has side-effects...)_